### PR TITLE
Adding pixell to conda-forge

### PR DESCRIPTION
**Pull request by iparask**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by msyriac** [comment]
I'm happy to be listed as a maintainer.

**Comment by JBorrow** [comment]
Happy to be a maintainer.

**Comment by iparask** [comment]
@conda-forge/help-python-c, ready for review!

File: recipes/pixell/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - {{ stdlib("c") }}
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+    - make
+    - cython
```
Comment (xhochy) [inline recipes/pixell/meta.yaml:28]: This should only be in `host`
```suggestion
```

File: recipes/pixell/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    - meson
+    - libopenblas * *openmp*
+    - libblas * *openblas
```
Comment (xhochy) [inline recipes/pixell/meta.yaml:31]: Should be in host
```suggestion
```

File: recipes/pixell/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    - healpy >=1.13
+    - matplotlib-base >=2.0
+    - pyyaml >=5.0
+    - pillow >=5.3.0,!=10.4.0
+    - pytest-cov >=2.6
+    - coveralls >=1.5
+    - pytest >=4.6
+    - ducc0 >=0.34.0
+    - numba >=0.54.0
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/pixell/meta.yaml:65]: This should no longer be used, `numpy` has a `run_exports`
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/pixell/meta.yaml (hunk 1 above)
Comment (iparask) [reply recipes/pixell/meta.yaml]: The build fails with:
```
  ../meson.build:3:0: ERROR: Cython compiler 'cython' cannot compile programs
```

when it is not included.

File: recipes/pixell/meta.yaml (hunk 2 above)
Comment (iparask) [reply recipes/pixell/meta.yaml]: Moved to host

**Comment by iparask** [comment]
@conda-forge-admin, please restart ci

**Comment by iparask** [comment]
@xhochy things are ready for another review. Thank you for your help!

File: recipes/pixell/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    - pyfftw
+    - libxcrypt1  # [linux]
+  run:
+    - llvm-openmp  # [osx]
+    - libopenblas * *openmp*
+    - libblas * *openblas
+    - {{ pin_compatible("numpy") }}
+    - {{ pin_compatible("scipy") }}
+    - {{ pin_compatible("libopenblas") }}
+    - {{ pin_compatible("libblas") }}
```
Comment (xhochy) [inline recipes/pixell/meta.yaml:59]: These all have `run_exports` and thus `pin_compatible` should not be used (and actually breaks things)
```suggestion
    - {{ pin_compatible("scipy") }}
```

File: recipes/pixell/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    - python
+    - astropy
+    - healpy
+    - pyyaml
+    - h5py
+    - pillow
+    - pyfftw
+
+
+test:
```
Comment (xhochy) [inline recipes/pixell/meta.yaml:70]:
```suggestion
test:
  require:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
Seems like this is missing the right constraints on the `run` requirements:

```
+ pip check
pixell 0.26.1 requires coveralls, which is not installed.
pixell 0.26.1 requires cython, which is not installed.
pixell 0.26.1 requires ducc0, which is not installed.
pixell 0.26.1 requires numba, which is not installed.
pixell 0.26.1 requires pytest, which is not installed.
pixell 0.26.1 requires pytest-cov, which is not installed.
pixell 0.26.1 has requirement Pillow!=10.4.0,>=5.3.0, but you have pillow 10.4.0.
``

**Review by xhochy — APPROVED** [review]

**Pull request merged**